### PR TITLE
Feature/list of projects

### PR DESCRIPTION
**Pull request by aberracja**
@pjar, @lsolniczek  please check. It's shorter this time :)

### CODE REVIEW

File: app/assets/stylesheets/projects/edit.scss [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/* override rails_form */
+
+.text-field {
+	margin: 0;
+}
+
+.text-label {
+	margin: 15px 0 5px;
+}
```
Comment (pjar) [inline app/assets/stylesheets/projects/edit.scss:9]: PLease fix indentation for all files

File: app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -1,7 +1,9 @@
 class ProjectsController < ApplicationController
-  def index
-    @project = current_user.projects
-  end
+
+	def index
+    projects_to_decorate = current_user.projects
+    @projects = ProjectPresenter.wrap(projects_to_decorate)
+	end
```
Comment (pjar) [inline app/controllers/projects_controller.rb]: Please fix indentation.

File: app/presenters/project_presenter.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+	end
+
+	def currency_or_empty
+		@currency ||= model.billable.blank? ? "---" : model.currency
+	end
+
+	def model
+		__getobj__
+	end
+end
```
Comment (pjar) [inline app/presenters/project_presenter.rb:22]: Please fix indentation.

File: app/models/project.rb [hunk 4]
```diff
@@ -15,6 +16,10 @@ def add_user_to_project(user)
     end
   end
 
+  def project_members_list
+    self.users.map{|u| u.first_name}.join(", ")
+  end
```
Comment (pjar) [inline app/models/project.rb]: This should go to decorator / presenter.

File: app/views/projects/edit.html.slim [hunk 5]
```diff
@@ -1,17 +1,19 @@
...
+	p
+		- @project.users.each do |p|
+			= p.email
+			br/
+
+	p.text-label
+		| invite to project:
+	= form_tag :action => 'invite' do
+		= text_field :invite_to_project, :email, placeholder: 'example@email.com', class: 'text-field'
+		= submit_tag "Invite", class: 'button'
```
Comment (pjar) [inline app/views/projects/edit.html.slim]: Please fix indentation.

File: app/controllers/projects_controller.rb [hunk 6]
```diff
@@ -1,7 +1,9 @@
 class ProjectsController < ApplicationController
-  def index
-    @project = current_user.projects
-  end
+
+	def index
+    projects_to_decorate = current_user.projects
+    @projects = ProjectPresenter.wrap(projects_to_decorate)
```
Comment (lsolniczek) [inline app/controllers/projects_controller.rb]: I don't think you need this line ' projects_to_decorate = current_user.projects'
You can just @projects = ProjectPresenter.wrap(current_user.projects)

**Comment by aberracja** [comment]
done! can I merge already? @pjar

File: app/assets/stylesheets/users/users.scss [hunk 7]
```diff
@@ -0,0 +1,8 @@
+.users {
+  @import "index"
+}
+
+
+
+
+
```
Comment (pjar) [inline app/assets/stylesheets/users/users.scss]: Remove empty lines

File: app/controllers/invitations_controller.rb [hunk 8]
```diff
@@ -1,59 +1,59 @@
...
+  def create
+    @invitation = Invitation.new(invitation_params)
+    @invitation.sender = current_user
+    @invitation.accepted = false
+    if @invitation.save
+      flash[:notice] = "Thank you, invitation created."
+      InvitationMailer.confirmation_email(@invitation).deliver
+      redirect_to users_path
+    else
+      flash[:alert] = "something is wrong :/ Probably you used wrong email address. Try again"
```
Comment (pjar) [inline app/controllers/invitations_controller.rb]: We can precisely tell what was wrong for every field.

File: app/controllers/invitations_controller.rb [hunk 9]
```diff
@@ -1,59 +1,59 @@
...
-        end
-
-        if @user.save
-          @invitation.update_columns(accepted: true)
-          InvitationMailer.first_signup_data(@invitation).deliver
-          redirect_to root_path, notice: 'You have accepted an invitation and have signed up successfully.'
-        else
-          redirect_to root_path
-        end
+  def confirm_invitation
```
Comment (pjar) [inline app/controllers/invitations_controller.rb:26]: We could keep REST here - action should be #update

File: app/controllers/projects_controller.rb [hunk 10]
```diff
@@ -24,19 +25,19 @@ def edit
...
-        flash[:notice] = "You have updated your project successfully."
-        redirect_to projects_path
-      else
-        render :edit
-      end
+    if @project.update(project_params)
+      flash[:notice] = "You have updated your project successfully."
+      redirect_to projects_path
+    else
+      render :edit
```
Comment (pjar) [inline app/controllers/projects_controller.rb:33]: No feedback for user

File: app/models/invitation.rb [hunk 11]
```diff
@@ -4,14 +4,14 @@ class Invitation < ActiveRecord::Base
   has_one :recipient, class_name: 'User'
 
   validates :recipient_email,
-      presence: true,
+            presence: true,
       format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i, on: :create },
```
Comment (pjar) [inline app/models/invitation.rb]: Fix indentation

File: app/views/projects/index.html.slim [hunk 12]
```diff
@@ -1,32 +1,45 @@
...
+      h4
+        | Projects
+    <!-- Table -->
+    - if @projects.blank?
+      | Add some projects!
+    - else
+      table.table
+        tr
+          th
+            | name
```
Comment (pjar) [inline app/views/projects/index.html.slim]: Please change to inline -  th name is enough

**Comment by aberracja** [comment]
@pjar can I merge now?

**Pull request merged**